howso/engine/trainee.py

```diff
@@ -3833,6 +3854,182 @@ def __exit__(self, exc_type, exc_val, exc_tb) -> None:
             if error.status != 404:
                 raise
 
+    def get_contribution_matrix(
+        self,
+        features: Optional[dict[str, str]] = None,
+        robust: bool = True,
+        targeted: bool = False,
+        normalize: bool = False,
+        normalize_method: str = "relative",
+        abval: bool = False,
+        fill_diagonal: bool = True,
+        fill_diagonal_value: Union[float, int] = 1,
+    ) -> DataFrame:
+        """
+        Gets the Feature Contribution matrix.
+
+        Parameters
+        ----------
+        features : Optional[Iterable[str]]
+            An iterable of feature names. If features are not provided, then the default trainee
+            features will be used.
+        robust : bool, default True
+            Whether to use robust calcuations.
+        targeted : bool, default False
+            Whether to do a targeted re-analyze before each feature's contribution is calculated.
+        normalize : bool, default False
+            Whether to normalize the matrix row wise. If postive and negative values are present, the normalized values
+            will be between -1 and 1. If only positive values are present, then normalized values will be between 0 and
+            1.
+        normalize_method : str, default 'relative'
+            The normalization method. Allowed values include 'relative' and 'fractional'.
+
+            - 'relative': normalizes each row by dividing each value by the maximum absolute value in the row.
+            - 'fractional': normalizes each row by dividing each value by the sum of absolute values in the row.
```

Comment:
its kind of an open question what is clearer. From a relative perspective, it shouldn't change anything. But 1.) Dividing by the absolute bounds it to -1, 1. Also lets say we have a row that's -5, 1, 5. If we take normalize by the normal sum, then it would be -5, 1, 5. If we normalize by the absolute sum, then it would be -.45, 0.09, .45. I might add both until its clear which one is more useful, maybe we end up keeping both.